chain/ethhashlookup/eth_transaction_hash_lookup.go

```diff
@@ -29,29 +29,77 @@ var ddls = []string{
 }
 
 const (
-	insertTxHash = `INSERT INTO eth_tx_hashes
-	(hash, cid)
-	VALUES(?, ?)
-	ON CONFLICT (hash) DO UPDATE SET insertion_time = CURRENT_TIMESTAMP`
+	insertTxHash    = `INSERT INTO eth_tx_hashes (hash, cid) VALUES(?, ?) ON CONFLICT (hash) DO UPDATE SET insertion_time = CURRENT_TIMESTAMP`
+	getCidFromHash  = `SELECT cid FROM eth_tx_hashes WHERE hash = ?`
+	getHashFromCid  = `SELECT hash FROM eth_tx_hashes WHERE cid = ?`
+	deleteOlderThan = `DELETE FROM eth_tx_hashes WHERE insertion_time < datetime('now', ?);`
 )
 
 type EthTxHashLookup struct {
 	db *sql.DB
+
+	stmtInsertTxHash    *sql.Stmt
+	stmtGetCidFromHash  *sql.Stmt
+	stmtGetHashFromCid  *sql.Stmt
+	stmtDeleteOlderThan *sql.Stmt
 }
 
-func (ei *EthTxHashLookup) UpsertHash(txHash ethtypes.EthHash, c cid.Cid) error {
-	hashEntry, err := ei.db.Prepare(insertTxHash)
+func NewTransactionHashLookup(ctx context.Context, path string) (*EthTxHashLookup, error) {
```

Comment:
I moved this up to the top, there's some diff noise from that but `initStatements()` at the bottom is new in this function